R/ResultsDataModel.R

```diff
@@ -255,12 +255,16 @@ createResultsDataModel <- function(connection = NULL,
   if (connection@dbms == "sqlite" & schema != "main") {
     stop("Invalid schema for sqlite, use schema = 'main'")
   }
-
-  sql <- SqlRender::loadRenderTranslateSql(sqlFilename = "CreateResultsDataModel.sql",
-                                           packageName = utils::packageName(),
-                                           dbms = connection@dbms,
-                                           results_schema = schema)
-  DatabaseConnector::executeSql(connection, sql)
+  
+  sql <- SqlRender::readSql(system.file("sql/sql_server/CreateResultsDataModel.sql", 
```

Comment:
Again, this refactoring is unnecessary and will probably not be tested during development. Its not a major change but why add it here?

Reply:
So the reason is 

`SqlRender::loadRenderTranslateSql` has the argument dbms. The default value is "sql_server". But notice that in prior code there is dbms = connection@dbms . So when connection was to sqlite it was looking for code in the path sqlite instead of sql_server. 

I found it confusing as it is an implicit and not explicit. This was implicit, because we always use the sql_server as the core folder for SQL. 

This code change is to make sure we are explicit and consistent in our pattern everywhere in the package (i.e. not just in the code for this PR).

Reply:
> SqlRender::loadRenderTranslateSql has the argument dbms. The default value is "sql_server". But notice that in prior code there is dbms = connection@dbms . So when connection was to sqlite it was looking for code in the path sqlite instead of sql_server.

This is confusing to me; the behaviour of `loadRenderTranslateSql` should look for the `sqlite` folder first then the `sql_server` folder. The reason for this is that there may be code that is specific to `sqlite` for which the translation doesn't work. In which case, you would add folders and sql files specific for that db platform. See the[ common data model ](https://github.com/OHDSI/CommonDataModel/tree/master/inst/ddl/5.4)for an example of this.

Reply:
But i dont get that same error in develop branch - only in the code for this PR. i dont understand why

This works fine
https://github.com/OHDSI/CohortDiagnostics/blob/develop/extras/tests/TestCodeEunomiaPackageMode.R

Reply:
The problem you're having is an interaction between devtools and SqlRender's file loader.

This is caused by the way that devtools test works; the normal mode of testing calls an actual package installed in the system but devtools uses a `load_all()` function. This can be solved by using rstudio for tests or creating a shortcut to the `<PackageRoot>/inst/sql` folder: `<PackageRoot>/sql`. This is resolved in linux/mac systems in the `test/testthat/helpers.R` but the solution used there (creating an sql symbolic link/shortcut when the tests run) does not work on windows because of permissions issues on many user's platforms.